src/Formats/NumpyDataTypes.h

```diff
@@ -0,0 +1,123 @@
+#pragma once
+#include <cstddef>
+#include <Storages/NamedCollectionsHelpers.h>
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+}
+
+enum class NumpyDataTypeIndex
+{
+    Int8,
+    Int16,
+    Int32,
+    Int64,
+    UInt8,
+    UInt16,
+    UInt32,
+    UInt64,
+    Float32,
+    Float64,
+    String,
+    Unicode,
+};
+
+class NumpyDataType
+{
+public:
+    enum Endianness
+    {
+        LITTLE,
+        BIG,
+        NONE,
+    };
+    NumpyDataTypeIndex type_index;
```

Comment:
I think we don't need this field in base class. In `NumpyDataTypeUnicode/NumpyDataTypeString` in getTypeIndex() we can just return `NumpyDataTypeIndex::String/Unicode` and don't save it in constructor. We save it in `NumpyDataTypeInt/NumpyDataTypeFloat` because they can have different type index.

But it just my though, current way is also ok